### PR TITLE
Progression traitor is now optional

### PR DESCRIPTION
**Pull request by Cenrus**
## About The Pull Request

Traitors can now decide if they want to use progression based objectives instead of pursuing their default steal/assassinate/escape ones.

Added new uplink option named Challenge Mode. To try and counteract snowballing, it limits your starting TC to 5 and unlocks the uplinks objective menu, allowing you to gain back your lost TC with enough effort.

Ported some improvements to prog objectives from tg

## Why It's Good For The Game

Some traitors prefer the more aggressive style of progression objectives, and this adds back some lost features such as Romerol and the battlecruiser as final objectives for them to complete.

## Changelog

:cl:
add: Traitors now have the option to do progression objectives
/:cl:

### CODE REVIEW

**Comment by BrotherHangyul** [comment]
_Might_ be beneficial to lower the price for Challenge mode to something like 15-18 TC, just so the traitor can still get basic tools. It's still possible to complete the objectives with just station tools, yes, but having access to something like a doorjack / EMP / just a better toolset might make it slightly more enjoyable. Don't take what I say too seriously though, I'm bad at balance.

File: code/modules/uplink/uplink_items/bundle.dm [hunk 1]
```diff
@@ -52,8 +52,37 @@
...
-/datum/uplink_item/bundles_tc
+/datum/uplink_item/bundles_tc/contractor
 	name = "Contractor Bundle"
 	desc = "A box containing everything you need to take contracts from the Syndicate. Kidnap people and drop them off at specified locations for rewards in the form of Telecrystals (Usable in the provided uplink) and Contractor Points."
 	item = /obj/item/storage/box/syndicate/contract_kit
+	progression_minimum = 90 MINUTES
+	cost = 20
+	
+/datum/uplink_item/bundles_tc/progression
+	name = "Challenge Mode"
```
Comment (Cenrus) [inline code/modules/uplink/uplink_items/bundle.dm:63]: I couldn't think of a better name. sorry.

**Review by Tokorizo — CHANGES_REQUESTED** [review]
I have a few issues with this PR, mostly the balance behind prog tots being an option route when we already have something that fills that niche called Contractors.

Prog tot is a bit of a messy snowball, depending on the amount of crew around and security, its incredibly easy to snowball infamously and TC by killing station pets and breaking into departments with no resistance. Id like to see some discussion on the github before I go forward with this, but the TLDR is simply; We have a niche for this, its called contractors.

**Comment by Zonespace27** [comment]
Contractor Shill/Creator/Enthusiast here, i would highly appreciate more content for contractors instead of this, which seems... to fall in pretty much the same category, as said above.

**Comment by Cenrus** [comment]
I understand your concerns about the snowballing and I'm starting to doubt if traitor progression is actually balanced myself. I'm closing this for now, and I'll think of other ways to implement final objectives back into the game

**Pull request closed without merging**